### PR TITLE
Move default vcr secrets to config/secrets.default.yml

### PR DESCRIPTION
**Pull request by agrare**

### CODE REVIEW

File: lib/manageiq/providers/oracle_cloud/engine.rb [hunk 1]
```diff
@@ -6,6 +6,11 @@ class Engine < ::Rails::Engine
 
         config.autoload_paths << root.join('lib').to_s
 
+        initializer :append_secrets do |app|
```
Comment (Fryguy) [inline lib/manageiq/providers/oracle_cloud/engine.rb:9]: You _might_ have to have some sort of `:before => :something_rails_secrets` here to make sure it slots into the right initialization order.

Comment (Fryguy) [reply lib/manageiq/providers/oracle_cloud/engine.rb:9]: Would be cool if we could find a way to "auto-inject" this into every provider

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -3,12 +3,14 @@
...
+Dir[Rails.root.join("spec/shared/**/*.rb")].sort.each { |f| require f }
+Dir[File.join(__dir__, "support/**/*.rb")].sort.each { |f| require f }
 
 require "manageiq-providers-oracle_cloud"
 
 VCR.configure do |config|
   config.ignore_hosts 'codeclimate.com' if ENV['CI']
   config.cassette_library_dir = File.join(ManageIQ::Providers::OracleCloud::Engine.root, 'spec/vcr_cassettes')
+  config.define_cassette_placeholder("ocid1.user.oc1..aaaaaaaa") { Rails.application.secrets.oracle_cloud[:user_id] }
+  config.define_cassette_placeholder("ocid1.tenancy.oc1..aaaaaaaa") { Rails.application.secrets.oracle_cloud[:tenant_id] }
```
Comment (Fryguy) [inline spec/spec_helper.rb:15]: Interesting...I wonder if there's a way we can access this key...something like...

In secrets.defaults.yml
```yaml
---
test:
  oracle_cloud_defaults: &oracle_cloud_defaults
    user_id: ocid1.user.oc1..aaaaaaaa
    ...
  oracle_cloud:
     <<: *oracle_cloud_defaults
```

Then here

```ruby
  config.define_cassette_placeholder(Rails.application.secrets.oracle_cloud_defaults[:user_id]) do
    Rails.application.secrets.oracle_cloud[:user_id]
  end
```

Comment (Fryguy) [reply spec/spec_helper.rb:15]: This way overrides will be in the oracle_cloud section, but the defaults would still be accessible in the oracle_cloud_defaults section.

File: .gitignore [hunk 3]
```diff
@@ -11,6 +11,7 @@
 .rubocop-*
 /bundler.d/*
 !/bundler.d/.keep
+/config/secrets.yml
```
Comment (Fryguy) [inline .gitignore:14]: Note that when we blast this out, we might want this in the generator.

File: lib/manageiq/providers/oracle_cloud/engine.rb (hunk 1 above)
Comment (Fryguy) [reply lib/manageiq/providers/oracle_cloud/engine.rb:9]: > You might have to have some sort of :before => :something_rails_secrets here to make sure it slots into the right initialization order.

I think this is fine...I can't find anything in the default initializers where we would hook it differently.

File: spec/factories/ext_management_system.rb [hunk 4]
```diff
@@ -14,24 +14,9 @@
       # To re-record the cassette replace these values with real ones and add
       # config.define_cassette_placeholder("ocid1.user.oc1..aaaaaaaa") { "real value" }
       # to spec/spec_helper.rb
```
Comment (Fryguy) [inline spec/factories/ext_management_system.rb:16]: Can this comment go away in this PR?

Comment (agrare) [reply spec/factories/ext_management_system.rb:16]: :+1: yes

File: spec/spec_helper.rb (hunk 2 above)
Comment (Fryguy) [reply spec/spec_helper.rb:15]: For example locally I have this:

```ruby
[3] pry(main)> require "yaml"
[4] pry(main)> x = YAML.load("test:
  oracle_cloud_defaults: &oracle_cloud_defaults
    user_id: ocid1.user.oc1..aaaaaaaa
  oracle_cloud: *oracle_cloud_defaults")
=> {"test"=>{"oracle_cloud_defaults"=>{"user_id"=>"ocid1.user.oc1..aaaaaaaa"}, "oracle_cloud"=>{"user_id"=>"ocid1.user.oc1..aaaaaaaa"}}}
[5] pry(main)> x["test"]["oracle_cloud"]["user_id"] = "foo"
=> "foo"
[6] pry(main)> x
=> {"test"=>{"oracle_cloud_defaults"=>{"user_id"=>"ocid1.user.oc1..aaaaaaaa"}, "oracle_cloud"=>{"user_id"=>"foo"}}}
```

Comment (agrare) [reply spec/spec_helper.rb:15]: Yeah I like the defaults, don't like a second top-level group but what about
```
oracle_cloud:
  defaults: &oracle_cloud_defaults
    user_id: xxx
    tenant_id: xxx
  <<: *oracle_cloud_defaults
```

This way it is all under `Rails.application.secrets.oracle_cloud`

Comment (Fryguy) [reply spec/spec_helper.rb]: oh nice!

**Comment by Fryguy** [comment]
I could be wrong, but I don;t think the vcr casettes need to be rebuilt with this PR (I don't see any changes other than the usual timestamps)

File: spec/spec_helper.rb (hunk 2 above)
Comment (agrare) [reply spec/spec_helper.rb]: :( unfortunately when I have a `config/secrets.yml` it doesn't merge but rather overrides everything under `test.oracle_cloud` so the `defaults` get blown away

**Review by Fryguy — APPROVED** [review]

File: spec/spec_helper.rb [hunk 5]
```diff
@@ -3,12 +3,19 @@
...
 VCR.configure do |config|
   config.ignore_hosts 'codeclimate.com' if ENV['CI']
   config.cassette_library_dir = File.join(ManageIQ::Providers::OracleCloud::Engine.root, 'spec/vcr_cassettes')
+
+  config.define_cassette_placeholder(Rails.application.secrets.oracle_cloud_defaults[:user_id]) do
+    Rails.application.secrets.oracle_cloud[:user_id]
+  end
+  config.define_cassette_placeholder(Rails.application.secrets.oracle_cloud_defaults[:tenant_id]) do
+    Rails.application.secrets.oracle_cloud[:tenant_id]
+  end
```
Comment (Fryguy) [inline spec/spec_helper.rb:20]: Now that I look at it, should there be a `define_cassette_placeholder` for the public/private key as well?  I guess those aren't stored in the vcr casette?

Comment (agrare) [reply spec/spec_helper.rb:20]: Those aren't stored in the cassette so we don't need placeholders for them

**Comment by Fryguy** [comment]
Skipping backport to `morphy`, because it is already in the branch.

**Pull request merged**